### DISCUSSION TITLE
Keywords blocking doesn't work on Firefox and Brave

### DISCUSSION THREAD

**Discussion by Hellvard8** [proginosko/LeechBlockNG#483] [Q&A] [answered]
Hello there!

For some reason site blocking works  for me just fine, but keyword blocking, not so much. I have "Access your data for all websites" checked on both browsers. Block set is set to all day, every day. Is there some setting I missed?

Firefox version: 129.0.1
Brave version:  1.68.141

**Comment by proginosko** [comment]
Please export your options to a text file (Options > General > Export Options) and email it to me at leechblockng@proginosko.com so I can take a closer look. Also provide a test case: a scenario where there should be a block but isn't.

**Reply by Hellvard8** [reply]
Oh, one thing I forgot to mention: adding keywords to blocklists breaks it, allowing me to access sites which were previously blocked. There's not so much of a scenario to test. I just typed random keywords, to test this feature: godzilla, batman and pokemon. I can type them in google and enter sites which contain them. I've send you my settings as you asked.

**Reply by proginosko** [reply, marked as answer]
Thanks. I think you are perhaps misunderstanding how keyword blocking works. See FAQ entry:

https://www.proginosko.com/leechblock/faq/keywords/

If you want to block some sites all the time (regardless of keywords) and other sites only when keywords appear, you'll need to use _two_ block sets, e.g.:

**Block Set 1: Block Some Sites Always**
facebook.com
youtube.com

**Block Set 2: Block Any Site When Keywords Appear**
*
~godzilla
~pokemon

**Reply by Hellvard8** [reply]
Ok, seems that I misunderstood this faq entry. Everything works fine now, thanks a lot!
